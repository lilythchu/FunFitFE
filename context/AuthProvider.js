import React, {createContext, useContext, useState} from 'react';

const AuthContext = createContext();

const AuthProvider = ({children}) => {
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [profile, setProfile] = useState({});
  const [token, setToken] = useState();

  return (
    <AuthContext.Provider
      value={{
        isLoggedIn,
        setIsLoggedIn,
        profile,
        setProfile,
        token,
        setToken,
      }}>
      {children}
    </AuthContext.Provider>
  );
};

export const useLogin = () => useContext(AuthContext);

export default AuthProvider;
