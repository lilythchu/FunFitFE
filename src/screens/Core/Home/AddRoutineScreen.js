import React, {useState} from 'react';
import {
  ActivityIndicator,
  Text,
  View,
  ScrollView,
  TouchableOpacity,
  TextInput,
  Alert,
} from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import {ListItem} from 'react-native-elements';
import CustomInput from '../../../components/CustomInput';
import CustomButton from '../../../components/CustomButton';
import TimeInput from '../../../components/TimeInput';
import Chevron from '../../../components/Chevron';

import {useForm} from 'react-hook-form';
import {useNavigation} from '@react-navigation/native';
import {useLogin} from '../../../../context/AuthProvider';
import globalStyles from '../../../../styles/global';
import globalColors from '../../../../styles/colors';
import client from '../../../../api/client';

const AddRoutineScreen = () => {
  const {token} = useLogin();
  const navigation = useNavigation();
  const [expanded, setExpanded] = useState(false);
  const {control, handleSubmit} = useForm();
  const [number, setNumber] = useState(0);
  const [loading, setLoading] = useState(false);

  const onAddRoutine = data => {
    setLoading(true);
    for (let i = 0; i < number; i++) {
      timings[i][0] = data[`h${i}`];
      timings[i][1] = data[`min${i}`];
      timings[i][2] = data[`sec${i}`];
    }
    const body = {
      name: data.name,
      duration: [data.hdur, data.mindur, data.secdur],
      genre: data.genre,
      steps: steps,
      timings: timings,
    };
    client
      .post('/routine/newRoutine', body, {
        headers: {Authorization: `Bearer ${token}`},
      })
      .then(response => navigation.navigate('Routine'))
      .catch(error => {
        if (error.response.data === 'Please choose a different name') {
          Alert.alert("Oops", error.response.data);
        } else {
          Alert.alert("Oops", "Something went wrong")
        }
      })
      .finally(() => setLoading(false));
  };
  var myLoop = [];
  var steps = [];
  var timings = [];

  for (let i = 0; i < number; i++) {
    steps[i] = `Step ${i + 1}`;
    timings[i] = ['00', '00', '00'];
    myLoop.push(
      <View style={{flexDirection: 'row'}} key={i}>
        <TextInput
          style={{flex: 1, textAlign: 'center'}}
          placeholder={`Step ${i + 1}`}
          onChangeText={text => (steps[i] = text)}
        />
        <TimeInput control={control} name={i} />
      </View>,
    );
  }

  return (
    // <ScrollView
    //   style={globalStyles.scrollView}
    //   showsVerticalScrollIndicator={false}>
    <KeyboardAwareScrollView
    style={{ backgroundColor: "white" }}
    resetScrollToCoords={{ x: 0, y: 0 }}
    contentContainerStyle={globalStyles.scrollView}
    scrollEnabled={true}>
      {/* Navigation */}
      <Chevron navigation={navigation} color={globalColors.babyBlue} />

      {/* Title */}
      <Text style={[globalStyles.title, {fontSize: 21}]}>
        Create your own Routine
      </Text>

      {/* Form */}
      <CustomInput
        name="name"
        placeholder="Name"
        type="THIRD"
        leftIcon={<Text style={{fontSize: 16}}>Name</Text>}
        control={control}
        rules={{
          required: 'Name is required',
        }}
      />
      <CustomInput
        name="genre"
        type="THIRD"
        leftIcon={<Text style={{fontSize: 16}}>Genre</Text>}
        placeholder="Genre"
        control={control}
        rules={{
          required: 'Genre is required'
        }}
      />
      <View style={globalStyles.durationContainer}>
        <Text style={{fontSize: 16}}>Duration</Text>
        <TimeInput control={control} name="dur" />
      </View>

      <ListItem.Accordion
        style={{borderWidth: 1}}
        noIcon
        content={
          <ListItem.Content>
            <TextInput
              keyboardType="numeric"
              placeholder="Number of steps(< 10) ~ One digit input"
              onChangeText={num => {
                setNumber(num);
                setExpanded(!expanded);
              }}
            />
          </ListItem.Content>
        }
        isExpanded={expanded}>
        <View style={{alignItems: 'center', marginVertical: 10}}>{myLoop}</View>
      </ListItem.Accordion>

      {/* Button */}
      <CustomButton
        title="Create"
        onPress={handleSubmit(onAddRoutine)}
        type="SECOND"
        loading={loading}
      />
    {/* // </ScrollView> */}
    </KeyboardAwareScrollView>
  );
};

export default AddRoutineScreen;
