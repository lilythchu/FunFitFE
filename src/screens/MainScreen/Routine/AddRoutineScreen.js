import React, {useState} from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Text,
  View,
  ScrollView,
  TouchableOpacity,
  TextInput,
} from 'react-native';
import Entypo from 'react-native-vector-icons/Entypo';
import CustomInput from '../../../components/CustomInput';
import CustomButton from '../../../components/CustomButton';
import { useNavigation } from '@react-navigation/native';
import { useForm } from 'react-hook-form';
import { ListItem } from '@rneui/themed';
import { globalStyles } from '../../../../styles/global';
import { addRoutineURL } from '../../../../api/client';
import { useLogin } from '../../../../context/AuthProvider';
import axios from 'axios';
import globalColors from '../../../../styles/colors';

const AddRoutineScreen = () => {
  const {token} = useLogin();
  const navigation = useNavigation();
  const [expanded, setExpanded] = useState(false);
  const {control, handleSubmit} = useForm();
  const [number, setNumber] = useState(0);
  const [loading, setLoading] = useState(false);

  const onAddRoutine = data => {
    setLoading(true);
    const body = {
      "name": data.name,
      "duration": duration,
      "genre": data.genre,
      "steps": steps,
      "timings": timings,
    };
    axios
      .post(addRoutineURL, body, {headers : {"Authorization": `Bearer ${token}`}})
      .then(response => {
        setLoading(false);
        navigation.navigate('Routine');
      })
      .catch(error => {
        console.log(error);
      });
  }
  var duration = ['00', '00', '00'];
  var myLoop = [];
  var steps = [];
  var timings = [];

  for (let i = 0; i < number; i++) {
    steps[i] = `Step ${i + 1}`;
    timings[i] = ['00', '00', '00'];
    myLoop.push(
      <View style={{flexDirection: 'row'}}> 
        <TextInput
          style={{flex: 1, textAlign: 'center'}}
          placeholder={`Step ${i + 1}`}
          onChangeText={text => steps[i] = text}
        />
        <View style={styles.timingContainer}>
          <TextInput
            style={styles.textBox}
            keyboardType='numeric'
            placeholder='00'
            onChangeText={number => timings[i][0] = number}
          />
          <Text>:</Text>
          <TextInput
            style={styles.textBox}
            keyboardType='numeric'
            placeholder='00'
            onChangeText={number => timings[i][1] = number}
          />
          <Text>:</Text>
          <TextInput
            style={styles.textBox}
            keyboardType='numeric'
            placeholder='00'
            onChangeText={number => timings[i][2] = number}
          />
        </View>
      </View>
    )
  }

  return (
    <ScrollView style={globalStyles.scrollView} showsVerticalScrollIndicator={false}>
      {/* Navigation */}
      <TouchableOpacity
        style={globalStyles.backIcon}
        onPress={() => navigation.goBack()}>
        <Entypo name='chevron-left' size={32} color={globalColors.babyBlue} />
      </TouchableOpacity>

      {/* Title */}
      <Text style={[globalStyles.title, {fontSize: 21}]}>Creat your own Routine</Text>
      
      {/* Form */}
      <CustomInput 
        name="name"
        placeholder="Name"
        type='THIRD'
        leftIcon={<Text>Name</Text>}
        control={control}
        rules={{
          required: "Name is required",
        }}
      />
      <CustomInput 
        name="genre"
        type='THIRD'
        leftIcon={<Text>Genre</Text>}
        placeholder="Genre"
        control={control}
        rules={{
          required: "Genre is required",
        }}
      />
      <View style={{flexDirection: 'row', paddingBottom: 10, paddingLeft: 10, alignItems: 'center'}}>
        <Text>Duration</Text>
        <View style={styles.timingContainer}>
          <TextInput
            style={styles.textBox}
            keyboardType='numeric'
            placeholder='00'
            onChangeText={number => duration[0] = number}
          />
          <Text>:</Text>
          <TextInput
            style={styles.textBox}
            keyboardType='numeric'
            placeholder='00'
            onChangeText={number => duration[1] = number}
          />
          <Text>:</Text>
          <TextInput
            style={styles.textBox}
            keyboardType='numeric'
            placeholder='00'
            onChangeText={number => duration[2] = number}
          />
        </View>
      </View>

      <ListItem.Accordion
        style={{borderWidth: 1}}
        content={
          <ListItem.Content>
            <TextInput
              keyboardType='numeric'
              placeholder='Number of steps(< 10)'
              onChangeText={number => {
                setNumber(number);
                setExpanded(!expanded);
              }}
            />
          </ListItem.Content>
        }
        isExpanded={expanded}
      >
        <View style={{alignItems: 'center', marginVertical: 10}}>
          {myLoop}
        </View>
      </ListItem.Accordion>
      
      {/* Button */}
      { loading
        ? <ActivityIndicator size="large" style={globalStyles.activityIdicator} />
        : <CustomButton
            title="Create"
            onPress={handleSubmit(onAddRoutine)}
            type="SECOND"
          />
      }
    </ScrollView>
  )
}

export default AddRoutineScreen; 

const styles = StyleSheet.create({
  textBox: {
    borderWidth: 1,
    width: 30,
    margin: 5,
    textAlign: 'center',
  },
  timingContainer: {
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
})