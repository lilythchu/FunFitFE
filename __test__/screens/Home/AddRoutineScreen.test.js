import React from 'react';
import {render, screen} from '@testing-library/react-native';
import AuthProvider from '../../../context/AuthProvider';
import AddRoutineScreen from '../../../src/screens/Core/Home/AddRoutineScreen';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

describe('Add routine screen', () => {
  test('renders correctly', () => {
    render(
      <AuthProvider>
        <AddRoutineScreen />
      </AuthProvider>
    );
    expect(screen).toMatchSnapshot();
  })
})