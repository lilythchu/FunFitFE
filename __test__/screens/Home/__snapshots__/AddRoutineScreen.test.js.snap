// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add routine screen renders correctly 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "backgroundColor": "white",
      "padding": 20,
    }
  }
>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginBottom": 10,
          "marginLeft": 20,
          "marginTop": 40,
          "opacity": 1,
        }
      }
      testID="chevron"
    >
      <Text />
    </View>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#0e86d4",
            "fontSize": 24,
            "fontWeight": "bold",
            "margin": 10,
          },
          Object {
            "fontSize": 21,
          },
        ]
      }
    >
      Creat your own Routine
    </Text>
    <View>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              Name
            </Text>
          </View>
          <TextInput
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="Name"
            placeholderTextColor="#86939e"
            style={
              Object {
                "color": "#242424",
                "flex": 1,
                "fontSize": 15,
                "minHeight": 40,
                "textAlign": "center",
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              Genre
            </Text>
          </View>
          <TextInput
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="Genre"
            placeholderTextColor="#86939e"
            style={
              Object {
                "color": "#242424",
                "flex": 1,
                "fontSize": 15,
                "minHeight": 40,
                "textAlign": "center",
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingLeft": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        Duration
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": 70,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#86939e",
                  "flexDirection": "row",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
              <TextInput
                editable={true}
                keyboardType="numeric"
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholderTextColor="#86939e"
                style={
                  Object {
                    "color": "#242424",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <Text>
                  :
                </Text>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "margin": 5,
                }
              }
            />
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": 70,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#86939e",
                  "flexDirection": "row",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
              <TextInput
                editable={true}
                keyboardType="numeric"
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholderTextColor="#86939e"
                style={
                  Object {
                    "color": "#242424",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <Text>
                  :
                </Text>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "margin": 5,
                }
              }
            />
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": 70,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#86939e",
                  "flexDirection": "row",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
              <TextInput
                editable={true}
                keyboardType="numeric"
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholderTextColor="#86939e"
                style={
                  Object {
                    "color": "#242424",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "margin": 5,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#bcbbc1",
            "flexDirection": "row",
            "padding": 14,
          }
        }
        testID="padView"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
        >
          <TextInput
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="Number of steps(< 10)"
          />
        </View>
        <View
          style={
            Object {
              "paddingLeft": 16,
            }
          }
        />
        <View
          collapsable={false}
          style={
            Object {
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "maxHeight": "0%",
          "opacity": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginVertical": 10,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          Object {
            "marginTop": 10,
          },
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#68bbe3",
              "borderColor": "transparent",
              "borderRadius": 30,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 40,
              "marginHorizontal": 20,
              "marginVertical": 20,
              "padding": 8,
              "width": 375,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Create
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
